### ISSUE TITLE
Windows CLI prove -o failed to create file. Need to revoke the proof.

### ISSUE THREAD

**Issue by mhykes** [keybase/keybase-issues#2159]
I am attempting to add proofs using the Windows CLI.
I first added github with
```
keybase prove github
```
and walked through the process of creating a gist successfully.  However, it was a bit challenging as the output is taller than my CLI window, so I had to copy the output to the gist in two chunks.  Since the help for prove offers the -o [outfile.name] flag, I figured I would learn my lesson and do that with the next proof.

Unfortunately,
```
C:\Program Files (x86)\Keybase>keybase prove -o twitter.proof twitter
Your username on Twitter: mattofnowhere
Please publicly tweet the following, and don't delete it:
- INFO Writing proof to file 'twitter.proof'...
- INFO Written.
Check Twitter now? [Y/n] n
- ERROR Proof wasn't available; we'll keep trying

C:\Program Files (x86)\Keybase>dir
 Volume in drive C has no label.
 Volume Serial Number is C8EB-4C62

 Directory of C:\Program Files (x86)\Keybase

03/25/2016  03:43 PM    <DIR>          .
03/25/2016  03:43 PM    <DIR>          ..
03/11/2016  11:02 AM        29,225,840 keybase.exe
03/25/2016  03:43 PM             4,709 unins000.dat
03/25/2016  03:43 PM         1,214,832 unins000.exe
03/25/2016  03:43 PM            22,553 unins000.msg
               4 File(s)     30,467,934 bytes
               2 Dir(s)  124,636,549,120 bytes free
```

As you can see, while the command states that it created the file, the file is not in the directory.  I used a windows explorer search (Win10) to search my entire C: drive for the file and did not find it.  Without the contents of this file, I cannot complete the proof by posting it in my twitter account.

As it happens, this is the wrong twitter account to use, anyway, since it is not a public account.
Since I will never complete the proof, I now need to cancel the process, or revoke the proof.  I cannot find any sign of the command to do this in the output from keybase help.

Please let me know:
 1. how to make sure I get an actual output file from keybase proof -o
 2. how to revoke this proof

In case it helps, 
```
C:\Program Files (x86)\Keybase>keybase -v
keybase version 1.0.14
```

**Comment by zanderz** [comment]
`C:\Program Files (x86)\Keybase` is not user-writeable by default, which you can verify by writing a file there with some other application, e.g. `C:\Program Files (x86)\Keybase>echo foo > foo.txt`. It should work if you change to somewhere writeable, like `cd %USERPROFILE%\Documents`. 

The client should give some kind of indication that the file wasn't written, though, so we'll file that part as a bug. Thanks!

**Comment by mhykes** [comment]
Yes, of course, now that you mention it, Windows went with making the program directories hard to write to in recent versions.  I should have thought of that.

Thank you for suggesting ```keybase sigs```.  Unfortunately,
```
C:\Program Files (x86)\Keybase>keybase sigs list
1    1c2d1ea2...   self     2016-03-25 17:05:20 EDT   mhykes
2    6e8f1173...   subkey   2016-03-25 17:05:20 EDT   0121689c4f56052821538199266ebf1c07087c3a6fc596b59b1b610d27a187519c590a
3    4b50280e...   sibkey   2016-03-25 17:05:21 EDT   0120954c1ca870d6307b62078f73ff6286ad88940e138f7e4031af157e344341c9b60a
4    34ce6974...   subkey   2016-03-25 17:05:21 EDT   0121b87bb8c228f1abb603b3d7f3accc1a5cf5d75895679e8b8694a789d137836f500a
5    a19fecd1...   track    2016-03-25 17:59:33 EDT   andym84
6    6eb16b8e...   proof    2016-03-25 22:54:40 EDT   mhykes@github
7    f67b85a9...   proof    2016-03-25 23:27:32 EDT   mattofnowhere@twitter
8    4dcf6daa...   track    2016-03-28 21:39:59 EDT   max
9    5a9786ad...   track    2016-03-29 23:34:39 EDT   zanderz

C:\Program Files (x86)\Keybase>keybase sigs revoke mattofnowhere@twitter
- ERROR Signature matching query "mattofnowhere@twitter" does not exist.
```

Maybe it won't revoke it because the process of confirming it is incomplete?

**Comment by zanderz** [comment]
You have to use the id, as in `keybase sigs revoke f67b85a9`. Maybe the prompt could be more clear.

**Comment by mhykes** [comment]
That worked.  
Thank you, zanderz.

**Comment by zanderz** [comment]
Update: Check for your proof files in `%USERPROFILE%\AppData\Local\VirtualStore\Program Files (x86)\Keybase`
File writes may be redirected here due to [virtualization](https://support.microsoft.com/en-us/kb/927387), a security feature I wasn't really aware of.

**Comment by mhykes** [comment]
I confirm.  The file ```twitter.proof```, containing the content I needed to post to twitter, was indeed at ```%USERPROFILE%\AppData\Local\VirtualStore\Program Files (x86)\Keybase```

Perhaps the Windows build for keybase.exe could know to check there as a fallback?

Alternatively, on install, create Windows users a home data directory in ```%USERPROFILE%\Documents\Keybase``` and have the Keybase CMD shortcut start in ```%USERPROFILE%\Documents\Keybase``` , with the path to keybase.exe added to the path so we can still just type ```keybase [COMMAND] [PARAMS]```.  Then you can place an introductory ```readme.txt``` in ```%USERPROFILE%\Documents\Keybase``` where it will be easy to find and read.  The ```readme.txt``` could contain a quick FAQ and suggest first steps for setting up and using the system from the command line.

I'd be happy to help getting that set up, or testing and verifying anything along those lines before you placed it in the regular release.  Unfortunately, I don't know enough GO to be able to just check out your source, make and test the changes, and provide a pull request.  I wish I did, because I think keybase is a really great idea.

**Comment by mhykes** [comment]
On second consideration, ```%USERPROFILE%\Documents``` fails on some older versions of Windows, and due to localization.

According to [http://stackoverflow.com/questions/3492920/is-there-a-system-defined-environment-variable-for-documents-directory]:

> The path to that folder is stored in "HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\Shell Folders" under "Personal". You need registry access though.

**Comment by mhykes** [comment]
And it seems I don't know how to use the comment buttons...  I never intended to close or reopen this.  I must need to sleep.
